recipes/python/backup-restore/common.py

```diff
@@ -0,0 +1,271 @@
+def verify_vmware_discovery_status(baseurl, token, workload_type, vcenter_server, timeout=600):
+    """ This function verify the 'SUCESS' discovery status of vCenter """
+    print(f"Wait for vCenter Discovery :[{vcenter_server}]")
+    discovery_status = ''
+    end_time = time.time() + timeout
+    while time.time() < end_time:
+        time.sleep(30)
+        discovery_status = get_vmware_discovery_status(baseurl, token, workload_type, vcenter_server)
+        if discovery_status == 'SUCCESS':
+            print(f"vCenter added successfully:[{vcenter_server}]")
+            break
+    else:
+        print(f"Failed to verify vCenter:[{vcenter_server}] discovery with status:[{discovery_status}]")
+        sys.exit(1)
+    print(f"vCenter discovery successful:[{vcenter_server}] with status:[{discovery_status}]")
+
+# Get asset info
+def get_asset_info(baseurl, token, workload_type, client):
+    """ This function return the asset info """
+    print(f"Get client asset info:[{client}]")
+    headers.update({'Authorization': token})
+    url = f"{baseurl}asset-service/workloads/{workload_type}/assets?"\
+                f"filter=commonAssetAttributes/displayName eq '{client}'"
+    status_code, response_text = rest_request('GET', url, headers)
+    validate_response(status_code, 200, response_text)
+
+    asset_id = response_text['data'][0]['id']
+    uuid = response_text['data'][0]['attributes']['instanceUuid']
+    exsi_host = response_text['data'][0]['attributes']['host']
+
+    print(f"Client asset Id:[{asset_id}]")
+    print(f"Client uuid Id:[{uuid}]")
+    print(f"Client exsi host:[{exsi_host}]")
+    return asset_id, uuid, exsi_host
+
+# Get StorageUnits
+def get_storage_units(baseurl, token):
+    """ This function return the storage unit name """
+    headers.update({'Authorization': token})
+    url = f"{baseurl}storage/storage-units"
+    status_code, response_text = rest_request('GET', url, headers)
+    validate_response(status_code, 200, response_text)
+    storage_unit_name = response_text['data'][0]['id']
+    is_instant_access_enable = response_text['data'][0]['attributes']['instantAccessEnabled']
+
+    if is_instant_access_enable:
+        print(f"Storage unit:[{storage_unit_name}] enabled for instant access")
+    else:
+        print(f"Storage unit:[{storage_unit_name}] disable for instant access")
+        raise Exception(f"Storage unit:[{storage_unit_name}] disabled for instant access")
+    return storage_unit_name
+
+# Create protection plan
+def create_protection_plan(baseurl, token, protection_plan_name, storage_unit_name):
+    """ This function create the protection plan """
+    print(f"Create protection plan:[{protection_plan_name}]")
+    headers.update({'Authorization': token})
+    url = f"{baseurl}servicecatalog/slos?meta=accessControlId"
+
+    cur_dir = os.path.dirname(os.path.abspath(__file__))
+    file_name = os.path.join(cur_dir, "create_protection_plan_template.json")
+    file_handle = open(file_name)
```

Comment:
Use `with` to ensure the file is closed promptly.

```suggestion
    with open(file_name, 'r') as file_handle:
```

Reply:
Used 'with' statement to open the file.